api/src/test/java/org/apache/iceberg/expressions/TestStringLiteralConversions.java

```diff
@@ -132,6 +132,38 @@ public void testStringToTimestampLiteral() {
         .isEqualTo(avroValue);
   }
 
+  @Test
+  public void testStringToTimestampLiteralWithMicrosecondPrecisionFromNanoseconds() {
+    // use Avro's timestamp conversion to validate the result
+    Schema avroSchema = LogicalTypes.timestampMicros().addToSchema(Schema.create(Schema.Type.LONG));
+    TimeConversions.TimestampMicrosConversion avroConversion =
+        new TimeConversions.TimestampMicrosConversion();
+
+    Literal<CharSequence> timestampStr = Literal.of("2017-08-18T14:21:01.123456789");
+    Literal<Long> timestamp = timestampStr.to(Types.TimestampType.microsWithoutZone());
+    long avroValue =
+        avroConversion.toLong(
+            LocalDateTime.of(2017, 8, 18, 14, 21, 1, 123456000).toInstant(ZoneOffset.UTC),
+            avroSchema,
+            avroSchema.getLogicalType());
+
+    assertThat((long) timestamp.value())
+        .as("Timestamp without zone should match UTC")
+        .isEqualTo(avroValue);
+  }
+
+  @Test
+  public void testStringToTimestampLiteralWithNanosecondPrecisionFromNanoseconds() {
+    // Not using Avro's timestamp conversion as it has no timestampNanos().
+    long expected = 1503066061123456789L;
```

Comment:
Test that Avro produces a value within 1 micro of this

Reply:
Done.